### PR TITLE
bnxt: Add support for rte_pmd_bnxt_set_vf_mac_anti_spoof

### PR DESCRIPTION
**Pull request by ajitkhaparde**
Add support for rte_pmd_bnxt_set_vf_mac_anti_spoof to enable VF MAC anti spoof.

Signed-off-by: Ajit Khaparde <ajit.khaparde@broadcom.com>

### CODE REVIEW

File: drivers/net/bnxt/bnxt_hwrm.c [hunk 1]
```diff
@@ -2113,3 +2113,19 @@ int bnxt_hwrm_func_bw_cfg(struct bnxt *bp, uint16_t vf,
 
 	return rc;
 }
+
+int bnxt_hwrm_func_cfg_set_vf_mac_spoof(struct bnxt *bp, uint16_t vf,
```
Comment (stephen-hurd) [inline drivers/net/bnxt/bnxt_hwrm.c]: This should just be something like bnxt_hwrm_func_cfg_set_flags() and not take a flags argument.

File: drivers/net/bnxt/rte_pmd_bnxt.c [hunk 2]
```diff
@@ -220,3 +220,54 @@ int bnxt_rcv_msg_from_vf(struct bnxt *bp, uint16_t vf_id, uint16_t type,
...
+		return 0;
+
+	func_flags = bp->pf.vf_info[vf].func_cfg_flags;
+
+	if (on)
+		func_flags |= HWRM_FUNC_CFG_INPUT_FLAGS_SRC_MAC_ADDR_CHECK;
+	else
+		func_flags &= ~HWRM_FUNC_CFG_INPUT_FLAGS_SRC_MAC_ADDR_CHECK;
+
+	rc = bnxt_hwrm_func_cfg_set_vf_mac_spoof(bp, vf, func_flags);
```
Comment (stephen-hurd) [inline drivers/net/bnxt/rte_pmd_bnxt.c]: This should just set the flags directly, then call the set flags function.  Unfortunately, the return code can be in response to any of the flag bits, not just the one we're changing.

**Review by stephen-hurd — CHANGES_REQUESTED** [review]
More generic flag setting is possible.

**Comment by ajitkhaparde** [comment]
On Mon, Feb 6, 2017 at 4:09 PM, Stephen Hurd <notifications@github.com>
wrote:

> *@stephen-hurd* requested changes on this pull request.
>
> More generic flag setting is possible.
> ------------------------------
>
> In drivers/net/bnxt/bnxt_hwrm.c
> <https://github.com/Broadcom/dpdk/pull/23#pullrequestreview-20381988>:
>
> > @@ -2113,3 +2113,19 @@ int bnxt_hwrm_func_bw_cfg(struct bnxt *bp, uint16_t vf,
>
>  	return rc;
>  }
> +
> +int bnxt_hwrm_func_cfg_set_vf_mac_spoof(struct bnxt *bp, uint16_t vf,
>
> This should just be something like bnxt_hwrm_func_cfg_set_flags() and not
> take a flags argument.
> ------------------------------
>
> In drivers/net/bnxt/rte_pmd_bnxt.c
> <https://github.com/Broadcom/dpdk/pull/23#pullrequestreview-20381988>:
>
> > +
> +	if (on > 1)	
> +		return -EINVAL;
> +
> +	/* Prev setting same as new setting. */
> +	if (on == bp->pf.vf_info[vf].mac_spoof_en)
> +		return 0;
> +
> +	func_flags = bp->pf.vf_info[vf].func_cfg_flags;
> +
> +	if (on)
> +		func_flags |= HWRM_FUNC_CFG_INPUT_FLAGS_SRC_MAC_ADDR_CHECK;
> +	else
> +		func_flags &= ~HWRM_FUNC_CFG_INPUT_FLAGS_SRC_MAC_ADDR_CHECK;
> +
> +	rc = bnxt_hwrm_func_cfg_set_vf_mac_spoof(bp, vf, func_flags);
>
> This should just set the flags directly, then call the set flags function.
> Unfortunately, the return code can be in response to any of the flag bits,
> not just the one we're changing.
>
​Hmm.. I thought only the "new" flag would be responsible for the return
code.
So if the return status is not necessarily because of the new flag, then I
will modify it to avoid the flags argument and just set the flags directly
before calling the hwrm function.

New patch-set coming up.​

File: drivers/net/bnxt/rte_pmd_bnxt.c [hunk 3]
```diff
@@ -220,3 +220,56 @@ int bnxt_rcv_msg_from_vf(struct bnxt *bp, uint16_t vf_id, uint16_t type,
...
+		func_flags &= ~HWRM_FUNC_CFG_INPUT_FLAGS_SRC_MAC_ADDR_CHECK;
+
+	bp->pf.vf_info[vf].func_cfg_flags = func_flags;
+
+	rc = bnxt_hwrm_func_cfg_vf_set_flags(bp, vf);
+	if (!rc) {
+		if (on)
+			bp->pf.vf_info[vf].mac_spoof_en = 1;
+		else
+			bp->pf.vf_info[vf].mac_spoof_en = 0;
```
Comment (stephen-hurd) [inline drivers/net/bnxt/rte_pmd_bnxt.c:270]: Why not just bp->pf.vf_info[vf].mac_spoof_en = on;

**Review by stephen-hurd — APPROVED** [review]

**Comment by ajitkhaparde** [comment]
:-) yes. why not. Will fix up that.

**Pull request merged**